pkg/controller/machine/machine.go

```diff
@@ -623,6 +628,15 @@ func (c *Controller) ensureNodeOwnerRefAndConfigSource(providerInstance instance
 		if err != nil {
 			return fmt.Errorf("failed to update machine status: %v", err)
 		}
+	} else {
+		// If the machine has an owner Ref and is older than 10 Minutes, delete it to have it re-created by the MachineSet controller
+		if machine.OwnerReferences != nil && !(c.joinClusterTimeout.Nanoseconds() == 0) && time.Since(machine.CreationTimestamp.Time) > c.joinClusterTimeout {
```

Comment:
Why not making `joinClusterTimeout` a pointer to express that it's an optional parameter?